internal/nimparser/nimparser.go

```diff
@@ -127,75 +127,92 @@ func MatchProfiles(modelSpec appsv1alpha1.ModelSpec, manifest NIMManifest, disco
 	var selectedProfiles []string
 
 	for hash, profile := range manifest {
-		if modelSpec.Precision != "" && profile.Tags["precision"] != modelSpec.Precision {
-			// continue with the next profile in the manifest
+		// Check precision, tensor parallelism, and QoS profile
```

Comment:
Hopefully we can find a place where nim team defines the keys -- if there are any updates we will know

Reply:
@slu2011 currently we have to rely on NIM repository.